components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/executors/NotificationExecutor.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.apimgt.core.executors;
+
+import org.apache.juli.logging.Log;
+import org.apache.juli.logging.LogFactory;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.wso2.carbon.apimgt.core.configuration.models.APIMConfigurations;
+import org.wso2.carbon.apimgt.core.exception.NotificationException;
+import org.wso2.carbon.apimgt.core.impl.Notifier;
+import org.wso2.carbon.apimgt.core.impl.NotifierConstants;
+import org.wso2.carbon.apimgt.core.template.dto.NotificationDTO;
+import org.wso2.carbon.apimgt.core.util.APIUtils;
+
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+
+/**
+ * Used to execute notification threads.
+ */
+public class NotificationExecutor {
+
+    private static final Log log = LogFactory.getLog(NotificationExecutor.class);
+    ExecutorService executor = Executors.newFixedThreadPool(10);
```

Comment:
Please check whether we should make the number configurable

Reply:
discussed this pool value earlier and came up with this fixed value as a solution for now.

Reply:
Ok.